packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -664,15 +664,15 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
   /// [FocusScopeNode.setFirstFocus].
   void unfocus({ bool focusPrevious = false }) {
     assert(focusPrevious != null);
-    if (!hasFocus && (_manager != null && _manager._nextFocus != this)) {
+    if (!hasFocus && (_manager == null || _manager._markedForFocus != this)) {
       return;
     }
     if (!hasPrimaryFocus) {
       // If we are in the focus chain, but not the primary focus, then unfocus
       // the primary instead.
       _manager?.primaryFocus?.unfocus(focusPrevious: focusPrevious);
     }
```

Comment:
Maybe not entirely related to this issue, but why is `_manager?.primaryFocus?.unfocus(focusPrevious: focusPrevious);` called when `hasFocus` is `true` and `hasPrimaryFocus` is `false`? The doc of `unfocus` states:
> Has no effect on nodes that return true from [hasFocus], but false from [hasPrimaryFocus].

Either the doc or the implementation seems to be wrong.

Reply:
Looks like the docs are wrong.  Thanks for pointing this out!